src/inventory.cpp

```diff
@@ -271,6 +271,29 @@ std::string ItemStack::getShortDescription(const IItemDefManager *itemdef) const
 	return desc;
 }
 
+f32 ItemStack::getRange(const IItemDefManager *itemdef) const
+{
+    std::string sRange = metadata.getString("range");
+    f32 range = -1;
+    if (sRange.empty())
+        range = getDefinition(itemdef).range;
+    if (!sRange.empty())
+    {
+        try {
+            //This is the best way (until otherwise said) to convert a string into a float.
+            //std::stof gives a compilation error along the lines of: namespace 'std' does not contain 'mystof'
+			f32 _range = std::strtof(sRange.c_str(), NULL);
+            if (_range != 0 || sRange == "0")
```

Comment:
This looks like a weird way to do validation:

* You will reject something like `0e42` or `0.0`, despite it being a valid float literal.
* It is not clear that the `_range != 0` is actually checking for an error.
* All of this is inside a `try`, implying that this was doing all the error handling (which it doesn't)

Also, the overall logic confuses me: You return range -1 for an "invalid" range, but you return the itemdef range for an empty range. Shouldn't you just default to the itemdef range in both cases?

Side note: Please format this to comply with the [code style guidelines](https://dev.minetest.net/Code_style_guidelines).

Reply:
> This looks like a weird way to do validation:
> 
> * You will reject something like `0e42` or `0.0`, despite it being a valid float literal.
> * It is not clear that the `_range != 0` is actually checking for an error.
> * All of this is inside a `try`, implying that this was doing all the error handling (which it doesn't)
> 
> Also, the overall logic confuses me: You return range -1 for an "invalid" range, but you return the itemdef range for an empty range. Shouldn't you just default to the itemdef range in both cases?
> 
> Side note: Please format this entire PR to comply with the [code style guidelines](https://dev.minetest.net/Code_style_guidelines).

I will remove the try-catch block if I can do it the way I'm about to explain. I want to create a function that returns false if the string is not a valid float before converting it. This was my original plan. The function would output like this.

`"Jdh76" -> false`
`"76.9" -> true`
`"836hdg" -> false`
`" " -> false`

Creating this kind of check function would be really simple. The problem is that I would have to place it in the inventory.cpp file which doean't seem like a very organized aproach but rather messy, so then where would I put it?